airgun/entities/oscapreports.py

```diff
@@ -0,0 +1,84 @@
+from time import sleep
+
+from airgun.entities.base import BaseEntity
+from airgun.navigation import NavigateStep
+from airgun.navigation import navigator
+from airgun.views.oscapreports import DeletereportsDialog
+from airgun.views.oscapreports import OSCAPReportsView
+
+
+class OSCAPReportsEntity(BaseEntity):
+    endpoint_path = '/compliance/arf_reports'
+
+    def search(self, value):
+        """Search for existing Compliance Reports."""
+        view = self.navigate_to(self, 'All')
+        return view.search(value)
+
+    def read_all(self):
+        """Read all values from Compliance Reports page."""
+        view = self.navigate_to(self, 'All')
+        return view.read()
+
+    def delete(self, search_pattern, cancel=False):
+        """Delete first Compliance Report based on search_pattern."""
+        view = self.navigate_to(self, 'All')
+        view.search(search_pattern)
+        view.table.row()['Actions'].widget.fill('Delete')
+        alert_message = self.browser.get_alert().text
+        self.browser.handle_alert(cancel=cancel)
+        view.flash.assert_no_error()
+        view.flash.dismiss()
+        return alert_message
+
+    def full_report(self, search_pattern, cancel=False):
+        """View Full Compliance Report."""
+        view = self.navigate_to(self, 'All')
+        view.search(search_pattern)
+        view.table.row()['Actions'].widget.fill('Full Report')
+        alert_message = self.browser.get_alert().text
+        self.browser.handle_alert(cancel=cancel)
+        view.flash.assert_no_error()
+        view.flash.dismiss()
+        return alert_message
```

Comment:
Looks like copy-paste from `delete`. As far as I can tell, there is no alert and no flash messages when clicking "Full Report".

Also, you probably want to return *something* from this report, not alert text.

Reply:
@mirekdlugosz thanks for your comments :) Fixed few issue you mentioned.

Reply:
I was thinking of covering `full report` method in other PR, but I guess I should add it in this one itself so I will move this PR to draft state till I finish it.

Reply:
@jameerpathan111 depends on you, really. If you can get some value out of this PR without `full_report` being implemented, I won't block it. But `full_report` method should raise NotImplementedError exception, or not exist at all (yet).

If there is nothing we could use without `full_report`, then yes, drafting and finishing later sounds like a good decision.